integration/volume/volume_test.go

```diff
@@ -0,0 +1,115 @@
+package volume
+
+import (
+	"context"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	volumetypes "github.com/docker/docker/api/types/volume"
+	"github.com/docker/docker/integration/internal/container"
+	"github.com/docker/docker/integration/internal/request"
+	"github.com/docker/docker/internal/testutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestVolumesList(t *testing.T) {
+	defer setupTest(t)()
+	client := request.NewAPIClient(t)
+	ctx := context.Background()
+
+	prefix, _ := getPrefixAndSlashFromDaemonPlatform()
+
+	id := container.Run(t, ctx, client, container.WithVolume(prefix+"foo"))
+
+	c, err := client.ContainerInspect(ctx, id)
```

Comment:
Does this need to create a container?  Could we use the volume create API instead? It should be much faster and lets us test the volume system independently from containers.

Reply:
Thanks @dnephin. I think it make sense to not create the container. I updated the PR. Also I merged  `TestVolumesCreate` and `TestVolumesList` into `TestVolumesCreateAndList`.

Reply:
I still see it creating a conatiner here. Is that change not pushed?

Reply:
Looks like it's needed to get a "volume in use" error below

Likely the container doesn't have to be _started_ though, only _created_ (which would make the test faster to run and cleanup)

Reply:
@thaJeztah @dnephin Thanks. The PR has been updated. Now the container is created but not started.